state/accountsDB.go

```diff
@@ -1113,17 +1113,25 @@ func (adb *AccountsDB) SnapshotState(rootHash []byte) {
 		stats.wg.Done()
 	}()
 
-	go func() {
-		printStats(stats, "snapshotState user trie", rootHash)
-
-		log.Debug("set activeDB in epoch", "epoch", epoch)
-		errPut := trieStorageManager.PutInEpoch([]byte(common.ActiveDBKey), []byte(common.ActiveDBVal), epoch)
-		handleLoggingWhenError("error while putting active DB value into main storer", errPut)
-	}()
+	go adb.markActiveDBAfterSnapshot(stats, rootHash, "snapshotState user trie", epoch)
 
 	adb.waitForCompletionIfRunningInImportDB(stats)
 }
 
+func (adb *AccountsDB) markActiveDBAfterSnapshot(stats *snapshotStatistics, rootHash []byte, message string, epoch uint32) {
+	printStats(stats, message, rootHash)
+
+	trieStorageManager := adb.mainTrie.GetStorageManager()
+	if trieStorageManager.IsClosed() {
+		log.Debug("will not set activeDB in epoch as the snapshot might be incomplete", "epoch", epoch)
```

Comment:
[comment]
Could've added extra explanation, like: 

```suggestion
		log.Debug("will not set activeDB in epoch as the snapshot might be incomplete, since trie storage manager is closed", "epoch", epoch)
```

Reply:
added extra explanation